Add options to output partition range

guofei opened the pull request:
## Summary

This pull request introduces three new options to improve the validation process when using the cassandra-data-migrator. These options are:

- `spark.cdm.tokenrange.partitionFile.appendOnDiff`
- `spark.cdm.tokenrange.partitionFile.input`
- `spark.cdm.tokenrange.partitionFile.output`

## Changes

1. The `spark.cdm.tokenrange.partitionFile.appendOnDiff` option allows the partition range to be outputted whenever there are differences, not just errors. This is beneficial as it allows us to focus on problematic ranges in subsequent validation runs, thereby reducing the overall execution time. This feature is backward compatible and only affects behavior when explicitly set to true.

2. The `spark.cdm.tokenrange.partitionFile.input` and `spark.cdm.tokenrange.partitionFile.output` options allow the specification of different input and output CSV files. This is useful in scenarios where the input and output files need to be different. These features are also backward compatible and only change behavior when the new options are used.

## Resolves

This PR addresses the issue raised in https://github.com/datastax/cassandra-data-migrator/issues/260

msmygit commented:
Thank you for your contributions! Someone on the core team will review the PR and get back to you. Appreciate in advance for your patience here

pravinbhat commented on src/main/java/com/datastax/cdm/job/SplitPartitions.java:
```diff
@@ -168,13 +168,32 @@ private static BufferedReader getfileReader(String fileName) {
+        }
+
+        if (!StringUtils.isAllBlank(propertyHelper.getString(KnownProperties.TOKEN_RANGE_PARTITION_FILE))) {
+            return propertyHelper.getString(KnownProperties.TOKEN_RANGE_PARTITION_FILE);
```
I would recommend we remove this property & completely replace with the `input` & `output` properties you have. I understand this breaks backward compatibility, but it will be more clear. In most scenarios when this feature is used, users just go with the default option, which will still continue to work.

pravinbhat commented:
Most changes look good, only a couple recommendations

- As noted above, I recommend completely removing the `partitionFile` option & go with the new separate `input`, `output` files instead
- Update all the other code (& test) to just go with `input`, `output` and `default`
- Update the properties [section here](https://github.com/datastax/cassandra-data-migrator/blob/main/src/resources/cdm-detailed.properties#L156) & add applicable comments

Thanks again for your contribution, looks really good!

guofei commented:
> Most changes look good, only a couple recommendations
> 
> * As noted above, I recommend completely removing the `partitionFile` option & go with the new separate `input`, `output` files instead
> * Update all the other code (& test) to just go with `input`, `output` and `default`
> * Update the properties [section here](https://github.com/datastax/cassandra-data-migrator/blob/main/src/resources/cdm-detailed.properties#L156) & add applicable comments
> 
> Thanks again for your contribution, looks really good!

Made the requested changes:

- Removed the `partitionFile` option 
- Updated `cdm-detailed.properties` with additional comments
- The README.md file has been updated to reflect these changes and provide clear instructions for the new options.

Please review the changes and let me know if there are any further modifications needed.
